lib/using_restricted_array.rb

```diff
@@ -1,61 +1,140 @@
+# Author: Farah Davoodi - Branches
+
 require_relative 'restricted_array.rb'
+require 'pry'
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n=length of array
+# Space complexity: O(1) no new arrays created, only one value returned
 def length(array)
-  raise NotImplementedError
+  if array[0] == nil
```

Comment:
In that case I'd return 0.  On the other hand if array == nil then you could return nil.